### ISSUE TITLE
[All platforms][Regression] Install omi package with [1.4.0-2]: omi process died after registering provider

### ISSUE THREAD

**Issue by LinfyZhai** [Microsoft/omi#413]
**ENV:** 
  
-   Clients: 
      omi64-sles10-t1(10.226.207.67)  [omi-1.4.0-2]                                        
      omi64-sles10-t2(10.226.207.233)[omi-1.3.0-105]

**Build:** 

- omi build 1.4.0-2: [\\\\redmond\wsscfs\OSTCData\Builds\omi\develop\1.4.0-2\Linux_ULINUX_1.0_x64_64_Release\openssl_1.0.0](url) 

- provider path: [\\\\ostcnfs-01\OSTCDATA\Team\v-zhenpy\providers\Linux_ULINUX_1.0_x64_64_Release\openssl_1.0.0\libTestClass_AllDMTFArrays.so
](url)

**Repro Steps:**
1.	Install omi package;
2.	Check omi process if exits with command “ps -ef|grep omi”;
3.	Register provider with command “/opt/omi/bin/omireg --hosting '@requestor@' -n root/cimv2 /root/libTestClass_AllDMTFArrays.so”;
4.	Check omi process if exits with command “ps -ef|grep omi”;

**Expect Result:**

After installed omi and registered provider, the omi process should **exits**.

**Details:**
- With build version **[omi-1.3.0-105]**. After installed omi and registered provider, the omi process still **exits**.
```
root@omi64-sles10-t2  # ps -ef|grep omi
root      3418     1  0 Oct15 ?        00:00:00 /sbin/dhcpcd -C -D -K -N -t 999999 -h omi64-sles10-t2 -c /etc/sysconfig/network/scripts/dhcpcd-hook eth0
root      5732     1  0 01:18 ?        00:00:00 /opt/omi/bin/omiserver --configfile=/etc/opt/omi/conf/omiserver.conf -d
omi       5754  5732  0 01:19 ?        00:00:00 /opt/omi/bin/omiengine --configfile=/etc/opt/omi/conf/omiserver.conf -d --socketpair 9
root      5757  5732  0 01:19 ?        00:00:00 /opt/omi/bin/omiagent 9 10 --destdir / --providerdir /opt/omi/lib --loglevel WARNING
root      5767  5323  0 01:20 pts/0    00:00:00 grep omi

root@omi64-sles10-t2  # rpm -qa|grep omi
omi-1.3.0-105

~
root@omi64-sles10-t2  # /opt/omi/bin/omireg --hosting '@requestor@' -n root/cimv2 /root/libTestClass_AllDMTFArrays.so
Created /opt/omi/lib/libTestClass_AllDMTFArrays.so
Created /etc/opt/omi/conf/omiregister/root-cimv2/TestClass_AllDMTFArrays.reg
/opt/omi/bin/omiserver: refreshed server

~
root@omi64-sles10-t2  # ps -ef|grep omi
root      3418     1  0 Oct15 ?        00:00:00 /sbin/dhcpcd -C -D -K -N -t 999999 -h omi64-sles10-t2 -c /etc/sysconfig/network/scripts/dhcpcd-hook eth0
root      5732     1  0 01:18 ?        00:00:00 /opt/omi/bin/omiserver --configfile=/etc/opt/omi/conf/omiserver.conf -d
omi       5922  5732  0 01:35 ?        00:00:00 /opt/omi/bin/omiengine --configfile=/etc/opt/omi/conf/omiserver.conf -d --socketpair 9
root      5924  5323  0 01:35 pts/0    00:00:00 grep omi

~
```

- With build version **[omi-1.4.0-2]**.  After installed omi and registered provider, the omi process **died**.
```
root@omi64-sles10-t1  # ps -ef|grep omi
root      3461     1  0 Oct15 ?        00:00:00 /sbin/dhcpcd -C -D -K -N -t 999999 -h omi64-sles10-t1 -c /etc/sysconfig/network/scripts/dhcpcd-hook eth0
root      9017     1  0 01:36 ?        00:00:00 /opt/omi/bin/omiserver --configfile=/etc/opt/omi/conf/omiserver.conf -d
omi       9018  9017  0 01:36 ?        00:00:00 /opt/omi/bin/omiengine --configfile=/etc/opt/omi/conf/omiserver.conf -d --logfilefd 3 --socketpair 9
root      9020  6440  0 01:36 pts/0    00:00:00 grep omi

~
root@omi64-sles10-t1  # rpm -qa|grep omi
omi-1.4.0-2

~
root@omi64-sles10-t1  # /opt/omi/bin/omireg --hosting '@requestor@' -n root/cimv2 /root/libTestClass_AllDMTFArrays.so
Created /opt/omi/lib/libTestClass_AllDMTFArrays.so
Created /etc/opt/omi/conf/omiregister/root-cimv2/TestClass_AllDMTFArrays.reg
/opt/omi/bin/omiserver: refreshed server

~
root@omi64-sles10-t1  # ps -ef|grep omi
root      3461     1  0 Oct15 ?        00:00:00 /sbin/dhcpcd -C -D -K -N -t 999999 -h omi64-sles10-t1 -c /etc/sysconfig/network/scripts/dhcpcd-hook eth0
root      9034  6440  0 01:37 pts/0    00:00:00 grep omi

~
```

**Comment by yakman2020** [comment]
Can we get some logs?  Is the omiserver segfaulting? If so, could we get a stack backtrace? IS the provider having issues?

**Comment by yakman2020** [comment]
That isn't a provider we build. Is it SCX?

**Comment by palladia** [comment]
Fixed with #414

**Comment by JumpingYang001** [comment]
Verified the fix on omi-1.4.0-3.
